pkg/commands/tls/custom/certificate/create.go

```diff
@@ -53,16 +63,41 @@ func (c *CreateCommand) Exec(_ io.Reader, out io.Writer) error {
 }
 
 // constructInput transforms values parsed from CLI flags into an object to be used by the API client library.
-func (c *CreateCommand) constructInput() *fastly.CreateCustomTLSCertificateInput {
+func (c *CreateCommand) constructInput() (*fastly.CreateCustomTLSCertificateInput, error) {
 	var input fastly.CreateCustomTLSCertificateInput
 
+	if c.certPath == "" && c.certBlob == "" {
+		return nil, fmt.Errorf("neither --cert-path or --cert-blob provided, one must be provided")
+	}
+
+	if c.certPath != "" && c.certBlob != "" {
+		return nil, fmt.Errorf("cert-path and cert-blob provided, only one can be specified")
+	}
+
 	if c.id != "" {
 		input.ID = c.id
 	}
-	input.CertBlob = c.certBlob
+
+	if c.certBlob != "" {
+		input.CertBlob = c.certBlob
+	}
+
+	if c.certPath != "" {
+		if path, err := filepath.Abs(c.certPath); err == nil {
+			if data, err := os.ReadFile(path); err == nil /* #nosec */ {
+				content := string(data)
+				input.CertBlob = content
+			} else {
+				return nil, fmt.Errorf("error reading cert-path: '%s' . Error: %q", path, err)
+			}
+		} else {
+			return nil, fmt.Errorf("cert file path invalid: '%s' . Error '%q'", path, err)
+		}
```

Comment:
I think it's more idiomatic to do this in reverse and handle the error cases first (also avoid the extra `else` + make the error message format consistent). The less nesting the better for readability...

```go
path, err := filepath.Abs(c.certPath)
if err != nil {
	return nil, fmt.Errorf("error parsing cert-path '%s': %q", path, err)
}

data, err := os.ReadFile(path) // #nosec	
if err != nil {
	return nil, fmt.Errorf("error reading cert-path '%s': %q", path, err)
}

input.CertBlob = string(data)
```